src/NServiceBus.Core/OpenTelemetry/Tracing/ActivityFactory.cs

```diff
@@ -27,7 +27,10 @@ public Activity StartIncomingActivity(MessageContext context)
         }
         else if (context.Headers.TryGetValue(Headers.DiagnosticsTraceParent, out var sendSpanId) && ActivityContext.TryParse(sendSpanId, null, out var sendSpanContext)) // otherwise directly create child from logical send
         {
-            activity = ActivitySources.Main.CreateActivity(name: ActivityNames.IncomingMessageActivityName, ActivityKind.Consumer, sendSpanContext);
+            // TryParse doesn't have an overload that supports changing the isRemote setting yet
```

Comment:
1. Shouldn't we consider to only support NET7 with Core v8 because we are so close to the RTM of NET7?
2. How would we know we can drop this code? I think it would be good if this code would be extracted to a method wrapped in a `#if NET6_0` block OR to have a explicit `#if !NET6_0` to then include `[Obsolete("Can be removed when targeting NET7", true)]`.

Reply:
@ramonsmits afaik we're going to the next LTS, not NET 7. I don't think the obsolete attribute can be used here since we're not exposing any API. I've raised #6587 to be considered for the next major version, that should ensure that we're getting to it (but if we don't it's also not the end of the world).